wallet-new/src/Cardano/Wallet/WalletLayer/Legacy.hs

```diff
@@ -113,7 +124,7 @@ pwlCreateWallet NewWallet{..} = do
     wId         <- migrate $ cwId wallet
 
     -- Get wallet or throw if missing.
-    maybeThrow (WalletNotFound wId) =<< pwlGetWallet wId
+    pwlGetWallet wId
```

Comment:
It's possible. Note that `MonadLegacyWallet ctx m` contains `MonadThrow m`.

Reply:
`impossible` was a poor word choice :D Is there an error that we can return from this function that is meaningful and potentially expected? eg are there well-typed inputs that predictably cause this function to throw?

Reply:
> are there well-typed inputs that predictably cause this function to throw?

You mean something like a wrong password?

Reply:
Yeah. `Map.insert` can't fail. But a SQL database insert might fail if a PK collision happens. And in the integration tests, we've already seen a wallet creation fail because the mnemonic was already taken. Returning `Either WalletFail Wallet` would be friendlier as the core part of the API (and a re-throwing variant can be added for convenience)